tests/Mapper.py

```diff
@@ -45,34 +55,63 @@ def tearDown(self):
 
     def testFileNameInterpretation(self):
         pfsParseTask = PfsParseTask(name='name')
-        site = 'S'
-        category = 'A'
-        visit = 4
-        arms = ['b', 'r', 'n', 'm']
-        minArmNum = 1
-        maxArmNum = 4
-        minSpectrographNum = 1
-        maxSpectrographNum = 4
-        for armNum in np.arange(minArmNum, maxArmNum + 1, 1):
-            for spectrograph in np.arange(minSpectrographNum, maxSpectrographNum + 1, 1):
-                ccd = spectrograph - 1
-                if int(armNum) == minArmNum + 1 or int(armNum) == minArmNum + 3:
-                    ccd += 4
-                elif int(armNum) == minArmNum + 2:
-                    ccd += 8
+        mapper = PfsMapper(
+            root=os.path.join(lsst.utils.getPackageDir('drp_stella_data'),'tests/data/PFS')
+        )
+        for armNum in np.arange(self.minArmNum, self.maxArmNum + 1, 1):
+            for spectrograph in np.arange(self.minSpectrographNum, self.maxSpectrographNum + 1, 1):
+                arm = self.arms[armNum-1]
+                dataId = {'spectrograph': spectrograph, 'arm': arm}
+                ccd = mapper.getDetectorId(dataId)
                 fileNameFormat = 'PF%1s%1s%06d%1d%1d.fits'
-                fileName = fileNameFormat % (site, category, visit, spectrograph, armNum)
+                fileName = fileNameFormat % (self.site, self.category, self.visit, spectrograph, armNum)
                 dirName = os.path.dirname( self.rawFiles )
                 fileNameWithDir = os.path.join( dirName, fileName )
                 info = pfsParseTask.getInfo( fileNameWithDir )
-                self.assertTrue(info[0]['site'] == site)
-                self.assertTrue(info[0]['category'] == category)
-                self.assertTrue(info[0]['visit'] == visit)
+                self.assertTrue(info[0]['site'] == self.site)
```

Comment:
Use `assertEqual(a, b)`, not `assertTrue(a == b)` wherever possible (better error messages)